shoop/core/models/_contacts.py

```diff
@@ -234,12 +239,18 @@ class Meta:
         verbose_name_plural = _('persons')
 
     def save(self, *args, **kwargs):
+        if self.first_name or self.last_name:
+            self.name = (" ".join([self.first_name, self.last_name])).strip()
         if self.user_id and not self.pk:  # Copy things
             user = self.user
             if not self.name:
                 self.name = user.get_full_name()
             if not self.email:
                 self.email = user.email
+            if not self.first_name and not self.last_name:
+                self.first_name = user.first_name
+                self.last_name = user.last_name
+
```

Comment:
Shouldn't the assigning of self.name come after this? Now its done before (lines 242-243) and self.name won't be set if first and last name are copied from the user.